packages/supabase_flutter/lib/src/flutter_go_true_client_options.dart

```diff
@@ -3,6 +3,13 @@ import 'package:supabase_flutter/supabase_flutter.dart';
 class FlutterAuthClientOptions extends AuthClientOptions {
   final LocalStorage? localStorage;
 
+  /// Callback for receiving state change updates from the auth subscription.
+  final void Function(AuthState data)? onAuthStateChange;
+
+  /// Callback for receiving errors from the auth state change subscription.
+  /// Both the error and the stacktrace will be given as arguments.
+  final Function? onAuthError;
+
```

Comment:
I don't think we want to introduce these. Let's remove them.

Reply:
Would you be okay with adding these in a different scope? The whole problem with the deeplink handling is the impossibility to catch the `passwordRecovery` event. To make this possible we need to have a listener before these events are pushed to the auth stream. 

This is an attempt to do so, as these listeners callbacks will be called even during initialization (= the phase where events like `passwordRecovery` are emitted).

Reply:
As long as we remove the duplicate initial session that gets fired after password recovery event, we should be good.

Reply:
Okay I'll revert the last commit! Do you have any tips on how to get the tests working again? I see quite a nice solution with mocking the method channels. Yet in AppLinks 6.0.0 the initial link handling turned into platform specific implementations, removing the ability to mock this in Dart 🤔

Reply:
Is it possible to create multiple mocks and test cases that mocks the behavior of web and non-web platforms?

Reply:
Hmm don't think so as the difference in usage is only dependent on the app_links package version. Tried to address this problem in latest commit, mocking both the event channel and the method channel with the initial link.

Reply:
Ending this comment chain here, not relevant to these changes